server_test.go

```diff
@@ -229,13 +239,16 @@ func TestServerNewMessageWithHeloContext(t *testing.T) {
 		message, heloRequest, heloResponse, helo := server.newMessage(), "heloRequest", "heloResponse", true
 		message.heloRequest, message.heloResponse, message.helo = heloRequest, heloResponse, helo
 		newMessage := server.newMessageWithHeloContext(message)
-		messages := server.messages.items
+		server.messages.append(newMessage)
```

Comment:
For what we are creating new message here? In our case we should just cover message creation from our method:

```go
func TestServerNewMessageWithHeloContext(t *testing.T) {
	t.Run("pushes new message into server.messages with helo context from other message, returns this message", func(t *testing.T) {
		server := &Server{messages: new(messages)}
		message, heloRequest, heloResponse, helo := server.newMessage(), "heloRequest", "heloResponse", true
		message.heloRequest, message.heloResponse, message.helo = heloRequest, heloResponse, helo
		newMessage := server.newMessageWithHeloContext(message)

		server.messages.RLock()
		messages := server.messages.items
		assert.Equal(t, heloRequest, newMessage.heloRequest)
		assert.Equal(t, heloResponse, newMessage.heloResponse)
		assert.Equal(t, helo, newMessage.helo)
		assert.Equal(t, newMessage, messages[0])
		assert.Equal(t, 1, len(messages))
		server.messages.RUnlock()
	})
}
```